src/Services/Interfaces/Repositories/IRepository.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Services.Interfaces.Repositories
+{
+    public interface IRepository<T> where T : class
+    {
+        IEnumerable<T> GetAll(Expression<Func<T, bool>>? filter = null, string? includeProperties = null);
+        T Get(Expression<Func<T, bool>> filter, string? includeProperties = null, bool tracked = false);
+        void Add(T entity);
+        void Remove(T entity);
+        void RemoveRange(IEnumerable<T> entities);
```

Comment:
Add an Update method and implement the basic use case of it. Be sure to allow overriding of it in later classes if a granular update is needed.

Reply:
Done